internal/field_decoder/field_decoder.go

```diff
@@ -294,3 +294,20 @@ func (d *FieldDecoder) WrapError(err error) FieldDecoderError {
 
 	return d.WrapError(d.decoder.WrapError(err))
 }
+
+func (d *FieldDecoder) WrapErrorAtOffset(err error, offset uint64) FieldDecoderError {
```

Comment:
Don't think it's intuitive that if the error is already wrapped, we don't update the offset - and if the error is already a decode error, we don't update the offset either. 

In general it's also a bit weird that we have to maintain byte positions manually again - the whole point of the refactor was that we could avoid doing this and our code would be simpler.  

@UdeshyaDhungana can you come up with some better options that solve this? I'm intentionally only mentioning problems here, not sharing solutions